wallet/wallet.c

```diff
@@ -1368,6 +1392,16 @@ bool wallet_init_channels(struct wallet *w)
 	return wallet_channels_load_active(w);
 }
 
+static enum channel_state_db get_state_channel_db(const char *dir, const char *type)
+{
+	for (int i = 0; i < CHANNEL_STATE_DB_SIZE; i++) {
+		struct channel_state_param state = channel_state_params[i];
+		if (streq(state.dir_key, dir) && streq(state.type_key, type))
+			return state.state;
+	}
```

Comment:
You can do one better. If you look at the above order you'll see that the `enum` has the structure you can use:

```
enum channel_state = IN_OFFERED;
if (streq(dir, "out"))
  channel_state += 2
if (streq(type, "fulfilled"))
  channel_state += 1
```

This works because of the order of your `enum`: least significant bit indicates state, most significant bit indicates direction. If you decide you want to do this I also suggest making this a bit more explicit by changing the above definition to this:

```
enum channel_state_db {
	IN_OFFERED = 0,
	IN_FULLFILLED = 1,
	OUT_OFFERED = 2,
	OUT_FULLFILLED = 4,
};
```

Reply:
Much much better! Thanks 😃

Reply:
Yep this is a great solution, thanks to suggesting it